modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSEntries.tsx

```diff
@@ -0,0 +1,483 @@
+	</>
+);
+
+interface IDropdownMenuProps {
+	headlessResources: Array<HeadlessResource>;
+	setHeadlessResourceValidationError: Function;
+	setSelectedHeadlessResource: Function;
+}
+
+const DropdownMenu = ({
+	headlessResources: initialHeadlessResources,
+	setHeadlessResourceValidationError,
+	setSelectedHeadlessResource,
+}: IDropdownMenuProps) => {
+	const [headlessResources, setHeadlessResources] = useState<
+		Array<HeadlessResource>
+	>(initialHeadlessResources || []);
+	const [query, setQuery] = useState('');
+
+	const onSearch = (query: string) => {
+		setQuery(query);
+
+		const regexp = new RegExp(query, 'i');
+
+		setHeadlessResources(
+			query
+				? initialHeadlessResources.filter(
+						({bundleLabel, name}: HeadlessResource) => {
+							return (
+								bundleLabel.match(regexp) || name.match(regexp)
+							);
+						}
+				  ) || []
+				: initialHeadlessResources
+		);
+	};
+
+	return (
+		<>
+			<ClayDropDown.Search
+				aria-label={Liferay.Language.get('search')}
+				onChange={onSearch}
+				value={query}
+			/>
+
+			<ClayDropDown.ItemList items={headlessResources} role="listbox">
+				{(item: HeadlessResource) => (
+					<ClayDropDown.Item
+						key={item.entityClassName}
+						onClick={() => {
+							setSelectedHeadlessResource(item);
+
+							setHeadlessResourceValidationError(false);
+						}}
+						roleItem="option"
+					>
+						<HeadlessResourceItem
+							headlessResource={item}
+							query={query}
+						/>
+					</ClayDropDown.Item>
+				)}
+			</ClayDropDown.ItemList>
+		</>
+	);
+};
+
+interface IFDSEntriesProps {
+	apiURL: string;
+	headlessResources: Array<HeadlessResource>;
+	namespace: string;
+}
+
+const FDSEntries = ({
+	apiURL,
+	headlessResources,
+	namespace,
+}: IFDSEntriesProps) => {
+	const headlessResourcesMapRef = useRef<Map<string, HeadlessResource>>(
+		new Map(
+			headlessResources.map((headlessResource) => [
+				headlessResource.entityClassName,
+				headlessResource,
+			])
+		)
+	);
+
+	type FDSEntry = {
+		entityClassName: string;
+	};
+
+	interface IProviderRendererProps {
+		itemData: FDSEntry;
+	}
+
+	const ProviderRenderer = ({itemData}: IProviderRendererProps) => {
+		const headlessResource = headlessResourcesMapRef.current.get(
+			itemData.entityClassName
+		);
+
+		return `${headlessResource?.name} (${headlessResource?.bundleLabel} ${headlessResource?.version})`;
+	};
+
+	interface IAddFDSEntryModalContentProps {
+		closeModal: Function;
+		loadData: Function;
+	}
+
+	const AddFDSEntryModalContent = ({
+		closeModal,
+		loadData,
+	}: IAddFDSEntryModalContentProps) => {
+		const [
+			selectedHeadlessResource,
+			setSelectedHeadlessResource,
+		] = useState<HeadlessResource>();
+		const [labelValidationError, setLabelValidationError] = useState(false);
+		const [
+			headlessResourceValidationError,
+			setHeadlessResourceValidationError,
+		] = useState(false);
+
+		const fdsEntryLabelRef = useRef<HTMLInputElement>(null);
+
+		const addFDSEntry = async () => {
```

Comment:
not in this PR, we will need to abstract the persistence out of the components when more complex cases arise (e.g. more than 1 fetch per operation. It's more than fine like this for now